.github/workflows/lint.yaml

```diff
@@ -0,0 +1,21 @@
+name: Lint checks
+on:
+  workflow_call:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - '**'
+  push:
+    branches:
+      - main
+
+jobs:
+  python:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+      - name: Ruff checks
```

Comment:
lol I think we both started similar type branches but I was doing it with Black. Did we decide not to use Black anymore? With black already configured in `pyproject.toml` we can just run it along side the current unit tests with poetry. Or we could probable add ruff to run in poetry instead?